src/SLCore/Service/Issue/Models/EffectiveIssueDetailsDto.cs

```diff
@@ -26,11 +26,11 @@
 namespace SonarLint.VisualStudio.SLCore.Service.Issue.Models;
 
 public record EffectiveIssueDetailsDto(
-    string ruleKey,
+    [JsonProperty("ruleKey")] string key,
```

Comment:
this name change is confusing

Reply:
It is needed for the implemenation of the IRuleDetails.
The EffectiveIssueDetailsDto and EffectiveRuleDetailsDto are very similar, which is why this interface was introduced. 
This also prevents code duplication of converting 99% similar Dtos to RuleInfo in the SLCoreRuleMetaDataProvider